File: src/main/java/aurocosh/divinefavor/common/core/handlers/BlockClickTracker.java
package aurocosh.divinefavor.common.core.handlers;

import aurocosh.divinefavor.common.custom_data.player.PlayerData;
import aurocosh.divinefavor.common.custom_data.player.data.interaction_handler.InteractionData;
import aurocosh.divinefavor.common.lib.math.Vector3i;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.event.entity.player.PlayerInteractEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.eventhandler.EventPriority;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

@Mod.EventBusSubscriber
public class BlockClickTracker {
    @SubscribeEvent(priority = EventPriority.LOWEST)
    public static void onPlayerLeftClickBlock(PlayerInteractEvent.LeftClickBlock event) {
        World world = event.getWorld();
        if(world.isRemote)
            return;
        EntityPlayer player = event.getEntityPlayer();
        InteractionData interactionData = PlayerData.get(player).getInteractionData();
        interactionData.recordLastClickedPosition(new Vector3i(event.getPos()));
    }

    public static boolean wasBlockLeftClicked(EntityPlayer player, BlockPos pos) {
        InteractionData interactionData = PlayerData.get(player).getInteractionData();
        return interactionData.wasPositionClicked(new Vector3i(pos));
    }
}

File: src/main/java/aurocosh/divinefavor/common/core/handlers/BlockClickTracker.kt
package aurocosh.divinefavor.common.core.handlers

import aurocosh.divinefavor.common.constants.ConstMisc
import aurocosh.divinefavor.common.lib.extensions.divineCustomData
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.math.BlockPos
import net.minecraftforge.event.entity.player.PlayerInteractEvent
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.eventhandler.EventPriority
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

@Mod.EventBusSubscriber(modid = ConstMisc.MOD_ID)
object BlockClickTracker {
    @SubscribeEvent(priority = EventPriority.LOWEST)
    fun onPlayerLeftClickBlock(event: PlayerInteractEvent.LeftClickBlock) {
        val world = event.world
        if (world.isRemote)
            return
        val player = event.entityPlayer
        val interactionData = player.divineCustomData.interactionData
        interactionData.recordLastClickedPosition(event.pos)
    }

    fun wasBlockLeftClicked(player: EntityPlayer, pos: BlockPos): Boolean {
        val interactionData = player.divineCustomData.interactionData
        return interactionData.wasPositionClicked(pos)
    }
}